cli/src/pcluster/aws/ec2.py

```diff
@@ -300,21 +304,22 @@ def get_official_image_id(self, os, architecture, filters=None):
         images = self._client.describe_images(Owners=[owner], Filters=filters, IncludeDeprecated=True).get("Images")
         if not images:
             raise AWSClientError(function_name="describe_images", message="Cannot find official ParallelCluster AMI")
+        return self._find_valid_official_image(images).get("ImageId")
 
-        def sort_by(image):
-            return ("0" if self._is_image_deprecated(image) else "1") + image["CreationDate"]
-
-        return max(images, key=sort_by).get("ImageId")
-
+    @AWSExceptionHandler.handle_client_exception
+    @Cache.cached
     def get_official_images(self, os=None, architecture=None):
         """Get the list of official images, optionally filtered by os and architecture."""
-        try:
-            owners = ["amazon"]
-            name = f"{self._get_official_image_name_prefix(os, architecture)}*"
-            filters = [{"Name": "name", "Values": [name]}]
-            return self.describe_images(ami_ids=[], owners=owners, filters=filters)
-        except ImageNotFoundError:
-            return []
```

Comment:
Does this deletion change the behavior of ParallelCluster CLI when the image cannot be found?

Reply:
No.
Both images created in `ACCOUNT_A` (see test description) are for `alinux2`, there are no images for `centos7`.
This is the test I performed:

```
.../python .../entrypoint.py list-official-images -r eu-west-1 --os centos7
{
  "images": []
}
```

`ImageNotFoundError` was thrown by `EC2Client.describe_images` which is not called anymore, because it wraps EC2 response into `ImageInfo` class, hiding information about `DeprecationTime`. EC2 client is now called directly, and it returns an empty list if there are no images. This is why I removed the exception handling.